### DISCUSSION TITLE
How to reduce delay in webrtc with nat ip

### DISCUSSION THREAD

**Discussion by krishgcek** [bluenviron/mediamtx#2650] [Questions]
### Question

Hello I am using webrtc with mediamtx and force it via nat ip and tcp.

After some time there is a delay about 20-25 seconds. How can I force the client to go to current time?

**Comment by aler9** [comment]
Hello, delay can be introduced in two points of your pipeline:

* by the video encoder, when hardware resources are not enough (are you encoding the video or just remuxing? you didn't provide any additional info)
* by the TCP transport protocol of WebRTC. Unfortunately TCP doesn't allow to discard packets, thus late packets can't be discarded but are piled up. UDP, on the other hand, perfectly allows that, and that's one of the reason why it's preferred to TCP in these real-time scenarios.

In my opinion, you'd better switch to UDP. The latest version of MediaMTX (1.3.0) includes a new simplified connectivity mechanism that allows to connect from both LAN and internet without using STUN or TURN servers. Instructions are here: https://github.com/bluenviron/mediamtx#connectivity-issues

**Reply by krishgcek** [reply]
I will update the version soon. I  solved the issue by using tcp for internet and UDP for lan for now.

[a comment by an automated account is not shown]
